centralstore/smartwrapper.go

```diff
@@ -113,8 +115,24 @@ func (w *SmartWrapper) WriteSpan(ctx context.Context, span *CentralSpan) error {
 // processSpans processes spans from the span channel. This is run in a
 // goroutine and runs indefinitely until cancelled by calling Stop().
 func (w *SmartWrapper) processSpans(ctx context.Context) {
-	for span := range w.spanChan {
-		err := w.BasicStore.WriteSpan(ctx, span)
+	for first := range w.spanChan {
+		var spans []*CentralSpan
+		spans = append(spans, first)
+
+	Remaining:
+		for len(spans) < WriteSpanBatchSize { // For control maximum size of batch
+			select {
+			case span, ok := <-w.spanChan:
```

Comment:
Will this have an opposite reaction where refinery will pause on processing spans when the rate of incoming spans is slow and it takes a while to reach `WriteSpanBatchSize`?